platform/ext/common/gcc/tfm_common_s.ld.template

```diff
@@ -253,6 +266,44 @@ SECTIONS
 
 #endif /* TFM_LVL != 1 */
 
+#ifndef TFM_MULTI_CORE_TOPOLOGY
+    /*
+     * Place the CMSE Veneers (containing the SG instructions) in a correctly
+     * aligned region so that the SAU can programmed to just set this region as
+     * Non-Secure Callable.
+     */
+    .gnu.sgstubs : ALIGN(32)
+    {
+        . = CMSE_VENEER_REGION_START_ALIGN;
+        Load$$LR$$LR_VENEER$$Base = ABSOLUTE(.);
+        *(.gnu.sgstubs*)
+#ifdef CMSE_VENEER_REGION_IN_CODE
+    } > FLASH AT> FLASH
+#else
+    } > VENEERS AT> VENEERS
+#endif
+
+    /*
+     * This is needed because the linker seems to place the veneers *after* the
+     * .gnu.sgstubs output section, so any end alignment applied inside it are
+     * placed before the veneers instead of after. Also, the end alignment seems
+     * to be ignored if not done in an output section, so this cannot be placed
+     * by itself after the .gnu.sgstubs section.
+     */
+    .sg_end_dummy : ALIGN(32)
+    {
+        . = CMSE_VENEER_REGION_END_ALIGN;
+        Load$$LR$$LR_VENEER$$Limit = ABSOLUTE(. - 1);
```

Comment:
@oyvindronningstad could you please, consider one more issue here: 
the _Limit address for the veneers' section is calculated, subtracting 1 from the current address pointer, which is different from the original implementation, where Limit is simply _start_ + sizeof (veneers section size)

That, also, appears to be the "standard" way of defining " veneer limit" addresses in TF-M; they do not consider the -1 in the calculations. See, for example, in target_cfg.c

```
    .veneer_base = (uint32_t)&REGION_NAME(Load$$LR$$, LR_VENEER, $$Base),
    .veneer_limit = (uint32_t)&REGION_NAME(Load$$LR$$, LR_VENEER, $$Limit),
```

I think this is because the veneers' region is only used in MPU configuration, not in SAU, IDAU.

Reply:
I see, wasn't aware. Updated now, thanks!

Reply:
We need to fix then (in a follow-up PR, I guess) this call:
https://github.com/NordicPlayground/trusted-firmware-m/blob/333c93e337a8f72fc4cf78a22e1247f4a65f6d9b/platform/ext/target/nordic_nrf/nrf5340/target_cfg.c#L185-L186
It's quite confusing that  "veneer limit" is understood differently than "non secure partition limit".